### ISSUE TITLE
[gather] L3_ADDEARIMM immediate overflow for 3D weight + 2D int64 index (MoE shape)

### ISSUE THREAD

**Issue by DevAyush22** [torch-spyre/torch-spyre#3499]
### What happened?

Compiling a MoE-style gather (`expert_w[expert_ids]`) with 3D fp16 weights and a 2D int64 index fails with BackendCompilerFailed. dxp_standalone aborts with `L3_ADDEARIMM :: be:be imm:-4194303 src0:0` — the L3 address immediate field overflows. The output tensor's per-core span is reported as 512 MB, which exceeds the 256 MB hardware limit, and the S-dim device stride is 4 MB, beyond the ±2 MB range of the 22-bit signed immediate.

### What did you expect to happen?

`torch.compile(lambda w, i: w[i], backend="inductor")` with MoE shapes should compile and run, producing output matching the CPU reference.

### How can we reproduce it?

```
import torch
import torch_spyre

E, D, F, B, S = 8, 512, 2048, 2, 128
expert_w = torch.rand(E, D, F, dtype=torch.float16, device="spyre")
expert_ids = torch.randint(0, E, (B, S), dtype=torch.int64).to("spyre")

torch.compile(lambda w, i: w[i], backend="inductor")(expert_w, expert_ids)
# → BackendCompilerFailed: dxp_standalone SIGABRT
```

Or via the test suite:

`SENCORES=1 python -m pytest tests/inductor/test_indirect_access_gather.py::TestGather::test_moe -s`

### Any environmental details we need to know?

Platform: Linux
Python 3.12.13
torch.compile(backend="inductor")
device: "spyre"

### Anything else we need to know?

The test is currently marked `@pytest.mark.skip` in tests/inductor/test_indirect_access_gather.py. 

### Relevant log output

```Shell
[WARNING] Cannot satisfy span limit for tensor buf0: coord=d1 still evaluates to 128 after splits.
[CRITICAL] buf0: per-core tensor span 512.000 MB (shape=[2, 128, 512, 2048],
  dtype=torch.float16, device_size=[128, 512, 32, 2, 64],
  splits={d0: 1, d1: 1, d2: 1, d3: 1}) exceeds hardware limit of 256.00 MB

DtException: Program verification failed for core 0 node 1_identity
Error message: Immediate value out of boundary in instruction:
L3_ADDEARIMM :: be:be imm:-4194303 src0:0
```

**Comment by DevAyush22** [comment]
cc: @shruthip-venkatesh @joyalbin @pradghos

**Comment by avery-blanchard** [comment]
Will be resolved by #2699

**Comment by Hemaprasannakc** [comment]
https://github.com/torch-spyre/torch-spyre/issues/3414 - similar issue
